### ISSUE TITLE
Logits processors documentation for v1

### ISSUE THREAD

**Issue by cpfiffer** [dottxt-ai/outlines#1562]
We don't currently document the logits processor, except to point to it in the [transformers](https://dottxt-ai.github.io/outlines/latest/reference/models/transformers/#using-logits-processors) doc.

The v1 docs should include an overview of how to use logit processors for advanced users.

Related: https://github.com/dottxt-ai/outlines/issues/1114
